pkg/logs/client/http/destination.go

```diff
@@ -56,13 +58,19 @@ func NewDestination(endpoint config.Endpoint, contentType string, destinationsCo
 // the error returned can be retryable and it is the responsibility of the callee to retry.
 func (d *Destination) Send(payload []byte) error {
 	ctx := d.destinationsContext.Context()
-	req, err := http.NewRequest("POST", d.url, strings.NewReader(string(payload)))
+
+	body, err := d.compression.compress(payload)
+	if err != nil {
+		return err
+	}
+	req, err := http.NewRequest("POST", d.url, bytes.NewReader(body))
 	if err != nil {
 		// the request could not be built,
 		// this can happen when the method or the url are valid.
 		return err
 	}
 	req.Header.Set("Content-Type", d.contentType)
+	d.compression.setHeader(&req.Header)
```

Comment:
I would define another abstraction to set the right headers as it's a different concern which compression can still implement though